src/metabase/models/metric.clj

```diff
@@ -65,7 +65,7 @@
 
 ;; ## Persistence Functions
 
-(defn create-metric
+(defn create-metric!
```

Comment:
Functions with side effects *aren't* safe in STM transactions. Code inside an STM transaction can get executed multiple times until the transaction succeeds.